packages/components/src/form-token-field/index.tsx

```diff
@@ -495,8 +496,19 @@ export function FormTokenField( props: FormTokenFieldProps ) {
 	}
 
 	function getSelectedSuggestion() {
+		const matchingSuggestions = getMatchingSuggestions();
+
 		if ( selectedSuggestionIndex !== -1 ) {
-			return getMatchingSuggestions()[ selectedSuggestionIndex ];
+			return matchingSuggestions[ selectedSuggestionIndex ];
+		}
+
+		if (
+			matchingSuggestions.length > 0 &&
+			incompleteTokenValue.length > 0 &&
+			typeof __experimentalValidateInput !== 'undefined' &&
```

Comment:
The `__experimentalValidateInput` prop is assigned a default value of `() => true`, and therefore this check will always be `true`.

I also don't understand the rationale for tying this new functionality to the `__experimentalValidateInput` — why is that prop necessary?

Reply:
Thanks for the review! I fixed the condition in a796d58. 

> I also don't understand the rationale for tying this new functionality to the `__experimentalValidateInput` — why is that prop necessary?

This is needed because if no validation is passed to `__experimentalValidateInput`, any string can be tokenized by pressing Enter key (or space if `tokenizeOnSpace` is `true`).` But you're right, we don't need that check because `addNewToken()` won't be executed if a matched suggestion is found. I deleted the unnecessary condition in [25052bd](https://github.com/WordPress/gutenberg/pull/42527/commits/25052bd0b4241760c748471549584bf1ae24c7e4).